core/net/rpl/rpl-ext-header.c

```diff
@@ -206,8 +208,10 @@ rpl_srh_get_next_hop(uip_ipaddr_t *ipaddr)
     }
   }
 
-  if(uip_next_hdr != NULL && *uip_next_hdr == UIP_PROTO_ROUTING
-      && UIP_RH_BUF->routing_type == RPL_RH_TYPE_SRH) {
+  if((uip_next_hdr != NULL && *uip_next_hdr == UIP_PROTO_ROUTING
+      && UIP_RH_BUF->routing_type == RPL_RH_TYPE_SRH) ||
+     (default_instance->current_dag->rank == ROOT_RANK(default_instance) &&
+      rpl_ns_get_node(dag, &UIP_IP_BUF->destipaddr) != NULL)) {
```

Comment:
Ah I see, this goes along with the other change below that suppresses SRH in case of a direct child. Makes sense. For a stricter check how about simply checking if the node returned by `rpl_ns_get_node()` has a parent field equal to the root?